homeassistant/components/input_timetable/__init__.py

```diff
@@ -0,0 +1,370 @@
+"""Support to set a timetable (on and off times during the day)."""
+import datetime
+import enum
+import logging
+import typing
+
+import voluptuous as vol
+
+import homeassistant
+from homeassistant.const import (
+    ATTR_EDITABLE,
+    ATTR_STATE,
+    CONF_ICON,
+    CONF_ID,
+    CONF_NAME,
+    SERVICE_RELOAD,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import collection, event as event_helper
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType, ServiceCallType
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "input_timetable"
+
+ATTR_TIME = "time"
+ATTR_TIMETABLE = "timetable"
+
+SERVICE_SET = "set"
+SERVICE_UNSET = "unset"
+SERVICE_RESET = "reset"
+SERVICE_RECONFIG = "reconfig"
+
+MIDNIGHT = datetime.time()
+
+
+class StateType(enum.Enum):
+    """Possible options for states."""
+
+    on = STATE_ON
+    off = STATE_OFF
+
+
+class StateEvent:
+    """State event properties (time, and state value)."""
+
+    def __init__(self, time, state):
+        """Initialize the object."""
+        self.time = time
+        self.state = state
+
+
+def remove_micros_and_tz(time):
+    """Remove microseconds and timezone from a time object."""
+    return time.replace(microsecond=0, tzinfo=None)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: cv.schema_with_slug_keys(
+            vol.Any(
+                {
+                    vol.Optional(CONF_NAME): cv.string,
+                    vol.Optional(CONF_ICON): cv.icon,
```

Comment:
We shouldn't make icon configurable, that is up to customisation or Lovelace

Reply:
Done